packages/template-retail-react-app/app/hooks/use-navigation.js

```diff
@@ -30,11 +32,7 @@ const useNavigation = () => {
          */
         (path, action = 'push', ...args) => {
             const locale = getLocaleByReference(site, localeShortCode)
-
-            const updatedHref = buildPathWithUrlConfig(path, {
-                locale: locale.alias || locale.id,
-                site: site.alias || site.id
-            })
+            const updatedHref = fillUrlTemplate(path, site.alias || site.id, locale.id || locale)
```

Comment:
```suggestion
            const updatedHref = fillUrlTemplate(path, site.alias || site.id, locale.alias || locale.id)
```
Is that right? Checking alias first and then the id?

Reply:
Yes, we want to use the Site alias first if it's defined in the App configuration. I removed the locale alias, since the alias concept doesn't exist in our App configuration.